src/View/Window.vala

```diff
@@ -440,14 +439,40 @@ namespace Marlin.View {
 
             content.loading.connect ((is_loading) => {
                 tab.working = is_loading;
-                update_top_menu ();
+                on_loading_or_active (content);
             });
 
             content.active.connect (() => {
-                update_top_menu ();
+                on_loading_or_active (content);
             });
 
             content.add_view (mode, location);
+
+            /* Need to have created a slot before changing tab to avoid a race where the topmenu can end up insensitive */
+            change_tab ((int)tabs.insert_tab (tab, -1));
+            tabs.current = tab;
+        }
+
+        private void on_loading_or_active (ViewContainer content) {
+            if (restoring_tabs || current_tab == null || content != current_tab) {
+                return;
+            }
+
+            /* Update browser buttons */
+            top_menu.set_back_menu (current_tab.get_go_back_path_list ());
```

Comment:
I really wish this could be kept on update_top_menu (), it got a bit confusing now.